src/m365/entra/commands/user/user-list.ts

```diff
@@ -52,37 +60,58 @@ class EntraUserListCommand extends GraphCommand {
   #initOptions(): void {
     this.options.unshift(
       {
-        option: "--type [type]",
-        autocomplete: ["Member", "Guest"]
+        option: '--type [type]',
+        autocomplete: EntraUserListCommand.allowedTypes
       },
-      { option: '-p, --properties [properties]' }
+      {
+        option: '-p, --properties [properties]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.type && !EntraUserListCommand.allowedTypes.some(t => t.toLowerCase() === args.options.type!.toLowerCase())) {
+          return `'${args.options.type}' is not a valid value for option 'type'. Allowed values are: ${EntraUserListCommand.allowedTypes.join(', ')}.`;
+        }
+
+        return true;
+      }
     );
   }
 
+  #initTypes(): void {
+    this.types.string.push('type', 'properties');
+  }
+
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     await this.showDeprecationWarning(logger, aadCommands.USER_LIST, commands.USER_LIST);
 
     try {
-      const selectProperties = args.options.properties ? args.options.properties : 'id,displayName,mail,userPrincipalName';
-      const allSelectProperties = selectProperties.split(',');
-      const propertiesWithSlash = allSelectProperties.filter(item => item.includes('/'));
+      let url = `${this.resource}/v1.0/users`;
 
-      const fieldExpand = propertiesWithSlash
-        .map(p => `${p.split('/')[0]}($select=${p.split('/')[1]})`)
-        .join(',');
+      if (args.options.properties) {
+        const selectProperties = args.options.properties;
+        const allSelectProperties = selectProperties.split(',');
+        const propertiesWithSlash = allSelectProperties.filter(item => item.includes('/'));
 
-      const expandParam = fieldExpand.length > 0 ? `&$expand=${fieldExpand}` : '';
-      const selectParam = allSelectProperties.filter(item => !item.includes('/'));
+        const fieldExpand = propertiesWithSlash
+          .map(p => `${p.split('/')[0]}($select=${p.split('/')[1]})`)
+          .join(',');
 
-      let filter: string = '';
-      try {
-        filter = this.getFilter(args.options);
+        const expandParam = fieldExpand.length > 0 ? `&$expand=${fieldExpand}` : '';
+        const selectParam = allSelectProperties.filter(item => !item.includes('/'));
+
+        url += `?$select=${selectParam}${expandParam}`;
       }
-      catch (ex: any) {
-        throw ex;
+
+      const unknownOptions: Options = this.getUnknownOptions(args.options);
```

Comment:
In fact, we cannot type this as `Options`. All properties declared in this interface will never be part of this object.